src/ray/raylet/node_manager.h

```diff
@@ -137,6 +137,20 @@ class NodeManager {
   std::unordered_map<ActorID, ActorRegistration> actor_registry_;
 };
 
+class TokenBucket {
```

Comment:
This class should be very clearly documented. From looking through this PR, I'm not totally sure what it does.